Edit GCHP emission diagnostic entries in HEMCO_Config.rc and HISTORY.rc files for consistency

yantosca opened the pull request:
### Name and Institution (Required)

Name: Bob Yantosca
Institution: Harvard + GCST

### Describe the update
This PR does the following:

1. Adds the following GCHP emission diagnostic entries that were present in `HEMCO_Diagn.rc.fullchem` but were missing in `HISTORY.rc.fullchem`:
  - EmisNO_fert (commented out, with note)
  - Added note for EmisSESQ_biogenic (commented out)
  - EmisTMB_Total
  - EmisTMB_Anthro
  - EmisTMB_Ship
  - InvCEDS_ALK6 (commented out)
  - InvCEDSship_ALK6 (commented out)
  - InvGTChlorine_HCl (commented out)
 
2. Adds the following GCHP emission diagnostic entries that were present in `HEMCO_Diagn.rc.carbon` but were missing in `HISTORY.rc.carbon`:
  - EmisOCS_Anthro
  - EmisOCS_Bioburn
  - EmisOCS_MissingOcean
  - EmisOCS_Ocean
  - EmisOCS_Total


3. Adds the following GCClassic emission diagnostic entry that was missing from `HEMCO_Diagn.rc.fullchem`
- InvCEDSship_ALK6 (commented out)

### Expected changes
These updates will restore missing GCHP emissions diagnostics.

### Related Github Issue

- See https://github.com/geoschem/gcpy/pull/357.  Missing emission entries were generated with the python code mentioned within.

lizziel reviewed (approved):
Looks good. If it is easy to do could you also add the updates to the files in the other run subdirectories (GEOS, etc)? If not, could you make a feature request to do that so I can track it?

yantosca commented:
@lizziel: I've updated the `run/GEOS/HEMCO_DiagnFile.rc` accordingly.  I didn't touch the `run/GEOS/HISTORY.rc` as I wasn't sure if they need to save out all those diagnostics.  

Also one potential bug fix: I changed the ALD2_PLANTDECAY emissions from category 3 to category 99, since in the HEMCO_Diagn.rc they use category 3 for anthropogenic transport.  Let me know if you think that is OK.

lizziel commented on CHANGELOG.md:
```diff
@@ -4,6 +4,15 @@ This file documents all notable changes to the GEOS-Chem repository starting in
+## [Unreleased] - TBD
+### Added
+- Added `#InvCEDSshipALK6` (commented out by default) to GCClassic `HEMCO_Diagn.rc.fullchem` and WRF `HEMCO_Diagn.rc` files
+- Added `EmisOCs*` diagnostics to `run/GCHP/HEMCO_Diagn.rc.templates/HEMCO_Diagn.rc.carbon` (these were missing)
```
This could be expanded to include also adding TMB emissions diagnostics to HEMCO_Diagn.rc.fullchem.

yantosca replied:
Now done in commit 7e0bcc1297a5627b0c629437415fd5611369e51d

lizziel commented on CHANGELOG.md:
```diff
@@ -4,6 +4,15 @@ This file documents all notable changes to the GEOS-Chem repository starting in
+
+### Changed
+- Updated GCHP template files `HEMCO_Diagn.rc.fullchem` and `HISTORY.rc.fullchem` so that the same emission diagnostics are requested in both
+- Changed `ALD2_PLANTDECAY` emissions category from 3 to 99 so as not to conflict with the anthropogenic transport sector.
```
Add that this is for GEOS-only

yantosca replied:
Now done in commit 7e0bcc1297a5627b0c629437415fd5611369e51d

lizziel reviewed (changes requested):
Thanks @yantosca! I will test out the GEOS file changes when I bring the version into GEOS, and run the plantdecay change by Viral. We can always change it back in geos/develop branch which will then come back into the standard model later.

lizziel commented:
@yantosca, could you update the PR description to include the changes made to the GEOS files, e.g. rearranged alphabetically, changed plant decay emissions diag category, added certain emissions, etc?

yantosca commented:
This PR will be integration-tested along with #2817.  Stay tuned for results.

yantosca commented:
All GEOS-Chem Classic integration tests passed:
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

CodeDir   : f29a0e8 GEOS-Chem update: Merge PR #2841 (UVFlux in GCHP alldiags test)
GEOS-Chem : 5b0e7603f Merge PR #2841 (Restore UVFlux in GCHP alldiags test)
HEMCO     : bf423d2 Change version numbers to 3.11.0
Cloud-J   : 1dff6fe Merge pull request #31 from geoschem/release/8.0.2
HETP      : 2a99b24 Merge pull request #2 from geoschem/bugfix/initialize_local_variables

Using 24 OpenMP threads
Number of execution tests: 42

Submitted as SLURM job: 13372474
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
All tests were zero-diff w/r/t 14.6.0 except:
- APM 
  - parallel error(expected)
- fullchem_alldiags; 
  - KppTime differs (should have been fixed previously, will investigate)
  - Dev has extra HEMCO diagnostics (expected)
- fullchem_benchmark
  - Dev has extra HEMCO diagnostics (expected)
  
All GCHP integration tests passed:
```console
==============================================================================
GCHP: Execution Test Results

CodeDir       : 41f1dec GEOS-Chem update: Merge PR #2841 (UVFlux in GCHP alldiags test)
MAPL          : 6207fd00 Merge pull request #38 from geoschem/bugfix/intel_build_error_in_AbstractComponent
GMAO_Shared   : 4ddb3ec Merge pull request #2 from geoschem/feature/mapl-upgrade
ESMA_cmake    : ad5deba Added ecbuild as a submodule of ESMA_cmake
gFTL-shared   : 4b82492 Merge branch 'upstream_v1.5.0' into feature/v1.5.0
FMS           : 259759d Merge pull request #3 from geoschem/feature/update_gmao_libs
FVdycoreCubed : af42462 Merge PR #8 (Add PLEadv diagnostic for offline advection in GCHP)
geos-chem     : 5b0e7603f Merge PR #2841 (Restore UVFlux in GCHP alldiags test)
HEMCO         : bf423d2 Change version numbers to 3.11.0
yaFyaml       : 19afe50 Merge branch 'upstream_v1.0.4' into feature/v1.0.4
pFlogger      : 2c4b724 Merge branch 'upstream_v1.9.1' into feature/v1.9.1
Cloud-J       : 1dff6fe Merge pull request #31 from geoschem/release/8.0.2
HETP          : 2a99b24 Merge pull request #2 from geoschem/bugfix/initialize_local_variables

Number of execution tests: 12

Submitted as SLURM job: 13372497
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
All tests were zero-diff w/r/t 14.6.0 except:
- fullchem_alldiags:
  - Dev has extra HEMCO diags (expected)
  - Dev has UVFlux restored (expected)
- fullchem_benchmark
  - Dev has extra HEMCO diags (expected)
